driver-sync/src/test/functional/com/mongodb/client/JsonPoweredCrudTestHelper.java

```diff
@@ -1084,19 +1084,34 @@ BsonDocument getUploadResult(final BsonDocument collectionOptions, final BsonDoc
     BsonDocument getClientWatchResult(final BsonDocument collectionOptions, final BsonDocument rawArguments,
                                       @Nullable final ClientSession clientSession) {
         MongoCursor<ChangeStreamDocument<Document>> cursor = mongoClient.watch().iterator();
-        return new BsonDocument("ok", new BsonInt32(1));
+        //noinspection TryFinallyCanBeTryWithResources
```

Comment:
It's a shame that whether we use try-with-resources in this case (which I would prefer seeing here, but approving anyway) or not, we have to either use `//noinspection TryFinallyCanBeTryWithResources` or `@SuppressWarnings("try")`. The latter is a manifestation of how useless the `javac`'s _"warning: [try] auto-closeable resource cursor is never referenced in body of corresponding try statement"_ is, because we can only put `@SuppressWarnings("try")` on the whole method.

Reply:
Indeed, it is.